phoenix-core/src/main/java/org/apache/phoenix/coprocessor/TaskRegionObserver.java

```diff
@@ -0,0 +1,240 @@
+            stmt.setByte(1, taskType.getSerializedValue());
+            stmt.setTimestamp(2, ts);
+            stmt.setString(3, schemaName);
+            stmt.setString(4, tableName);
+            stmt.execute();
+            conn.commit();
+        } catch (SQLException e) {
+            throw new IOException(e);
+        }
+    }
+
+    /**
+     * Task runs periodically to clean up task of child views whose parent is dropped
+     *
+     */
+    public static class DropChildViewsTask extends TimerTask {
+        private RegionCoprocessorEnvironment env;
+        private long timeMaxInterval;
+
+        public DropChildViewsTask(RegionCoprocessorEnvironment env, long timeMaxInterval) {
+            this.env = env;
+            this.timeMaxInterval = timeMaxInterval;
+        }
+  
+        @Override
+        public void run() {
+            PhoenixConnection connForTask = null;
+            try {
+                String taskQuery = "SELECT " +
+                        PhoenixDatabaseMetaData.TASK_TS + ", " +
+                        PhoenixDatabaseMetaData.TENANT_ID + ", " +
+                        PhoenixDatabaseMetaData.TABLE_SCHEM + ", " +
+                        PhoenixDatabaseMetaData.TABLE_NAME +
+                        " FROM " + PhoenixDatabaseMetaData.SYSTEM_TASK_NAME +
+                        " WHERE "+ PhoenixDatabaseMetaData.TASK_TYPE + " = " + PTable.TaskType.DROP_CHILD_VIEWS.getSerializedValue();
+
+                connForTask = QueryUtil.getConnectionOnServer(env.getConfiguration()).unwrap(PhoenixConnection.class);
+                PreparedStatement taskStatement = connForTask.prepareStatement(taskQuery);
+                ResultSet rs = taskStatement.executeQuery();
+                while (rs.next()) {
+                    try {
+                        // delete child views only if the parent table is deleted from the system catalog
+                        Properties tenantProps = new Properties();
+                        if (rs.getString(3) != null) {
+                            tenantProps.setProperty(PhoenixRuntime.TENANT_ID_ATTRIB, rs.getString(3));
+                        }
+                        PhoenixConnection pconn = QueryUtil.getConnectionOnServer(tenantProps, env.getConfiguration()).unwrap(PhoenixConnection.class);
+
+                        MetaDataProtocol.MetaDataMutationResult result = new MetaDataClient(pconn).updateCache(pconn.getTenantId(),
+                                rs.getString(3), rs.getString(4), true);
+                        if (result.getMutationCode() != MetaDataProtocol.MutationCode.TABLE_ALREADY_EXISTS) {
+                            MetaDataEndpointImpl.dropChildViews(env, rs.getBytes(2), rs.getBytes(3), rs.getBytes(4));
+                        } else if (System.currentTimeMillis() < timeMaxInterval + rs.getTimestamp(1).getTime()) {
+                            // skip this task as it has not been expired and its parent table has not been dropped yet
```

Comment:
Maybe add an info log line here for this

Reply:
You can also format the log string with `String.format`.